src/cmds/qselect.c

```diff
@@ -393,7 +399,7 @@ main(int argc, char **argv, char **envp) /* qselect */
 
 	struct attropl *select_list = 0;
 
-	static char destination[PBS_MAXQUEUENAME+1] = "";
+	static char destination[MAXSERVERNAME+PBS_MAXQUEUENAME+1] = "";
```

Comment:
Man page for qselect says:

               Restricts selection to those jobs at the specified destination.

               The destination may take of one of the following forms:
                  <queue name>
                  Restricts selection to the specified queue at the default server.
                  @<server name>
                  Restricts selection to the specified server.
                  <queue name>@<server name>
                  Restricts  selection  to  the  specified  queue  at the specified
                  server.

so we need to add a +1 for the "@" sign. Can you also check if <server_name> would also include a port number, ala  <server_host>:<port_number>, and if so, the following might be better as defined in pbs_ifl.h:
#define PBS_MAXROUTEDEST        (PBS_MAXQUEUENAME + PBS_MAXSERVERNAME + PBS_MAXPORTNUM + 2) /* destination size */

Reply:
Thanks. Fixed. `MAXSERVERNAME` already contains enough space for server name and port number.